test/benchmarks/include/vikunja/bench/memory.hpp

```diff
@@ -0,0 +1,213 @@
+/* Copyright 2022 Simeon Ehrig
+ *
+ * This file is part of vikunja.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <alpaka/alpaka.hpp>
+
+#include <algorithm>
+#include <vector>
+
+namespace vikunja::bench
+{
+    template<typename TData>
+    class IotaFunctor
+    {
+    private:
+        TData const m_begin;
+        TData const m_increment;
+
+    public:
+        //! Functor for iota implementation with generic data type.
+        //!
+        //! \tparam TData Type of each element
+        //! \param begin Value of the first element.
+        //! \param increment Distance between two elements.
+        IotaFunctor(TData const begin, TData const increment) : m_begin(begin), m_increment(increment)
+        {
+        }
+
+        //! Writes the result of `begin + index * increment` to each element of the output vector.
+        //!
+        //! \tparam TAcc The accelerator environment to be executed on.
+        //! \tparam TElem The element type.
+        //! \param acc The accelerator to be executed on.
+        //! \param output The destination vector.
+        //! \param numElements The number of elements.
+        ALPAKA_NO_HOST_ACC_WARNING
+        template<typename TAcc, typename TIdx>
+        ALPAKA_FN_ACC auto operator()(TAcc const& acc, TData* const output, TIdx const& numElements) const -> void
+        {
+            static_assert(alpaka::Dim<TAcc>::value == 1, "The VectorAddKernel expects 1-dimensional indices!");
+
+            TIdx const gridThreadIdx(alpaka::getIdx<alpaka::Grid, alpaka::Threads>(acc)[0u]);
+            TIdx const threadElemExtent(alpaka::getWorkDiv<alpaka::Thread, alpaka::Elems>(acc)[0u]);
+            TIdx const threadFirstElemIdx(gridThreadIdx * threadElemExtent);
+
+            if(threadFirstElemIdx < numElements)
+            {
+                // Calculate the number of elements for this thread.
+                // The result is uniform for all but the last thread.
+                TIdx const threadLastElemIdx(threadFirstElemIdx + threadElemExtent);
+                TIdx const threadLastElemIdxClipped(alpaka::math::min(acc, numElements, threadLastElemIdx));
+
+                for(TIdx i(threadFirstElemIdx); i < threadLastElemIdxClipped; ++i)
+                {
+                    output[i] = m_begin + static_cast<TData>(i) * m_increment;
+                }
+            }
+        }
+    };
+
+
+    //! Allocates memory and initialises each value with `begin + index * increment`,
+    //! where index is the position in the output vector. The allocation is done with `setup.devAcc`.
+    //!
+    //! \tparam TData Data type of the memory buffer.
+    //! \tparam TSetup Fully specialized type of `vikunja::test::TestAlpakaSetup`.
```

Comment:
You are using British spelling above (`initiali_s_es`) but American spelling here (`speciali_z_ed`).

Reply:
I didn't know, that there is an `initialize` :sweat_smile: